index.js

```diff
@@ -149,6 +150,90 @@ class GithubScm extends Scm {
     }
 
     /**
+     * Adds the Screwdriver webhook to the Github repository
+     * @method _addWebhook
+     * @param  {Object}    config            Config object
+     * @param  {String}    config.scmUri     The SCM URI to add the webhook to
+     * @param  {String}    config.token      Service token to authenticate with Github
+     * @param  {String}    config.webhookUrl The URL to use for the webhook notifications
+     * @return {Promise}                     Resolve means operation completed without failure.
+     */
+    _addWebhook(config) {
+        const searchForHook = (scmInfo, url, page) => {
+            const getHooks = this.breaker.runCommand({
+                action: 'getHooks',
+                token: config.token,
+                params: {
+                    owner: scmInfo.owner,
+                    repo: scmInfo.repo,
+                    page,
+                    per_page: WEBHOOK_PAGE_SIZE
+                }
+            });
+            const filterForHook = getHooks.then((hooks) => {
+                let found = null;
+
+                hooks.some((hook) => {
+                    const currentHook = hoek.reach(hook, 'config.url');
+
+                    if (currentHook === url) {
+                        found = hook;
+                    }
+
+                    return found;
+                });
+
+                return found;
+            });
+
+            return Promise.all([
+                getHooks,
+                filterForHook
```

Comment:
But this is 2 levels. `Promise.all` is usually done when the 2 promises can be done independently. For example: https://github.com/screwdriver-cd/screwdriver/blob/5911bff2bc2d7b584634e402ab306b0f5a8f9843/plugins/pipelines/remove.js#L25-L28

I find it's weird to use `Promise.all` in this scenario, since you're basically calling `getHooks` twice.